pkg/services/mattermost.go

```diff
@@ -0,0 +1,100 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	texttemplate "text/template"
+
+	log "github.com/sirupsen/logrus"
+
+	httputil "github.com/argoproj-labs/argocd-notifications/pkg/util/http"
+)
+
+type MattermostNotification struct {
```

Comment:
I was under impression that we can just re-use Slack notification and templates but looks like not every field is supported. 

I see attachments in documentation but no blocks, users etc https://docs.mattermost.com/developer/message-attachments.html#attachment-options

@ryota-sakamoto can you please confirm that I understand it correctly?

Reply:
Basically mattermost and slack are similar but not really compatible, right?

Reply:
@alexmt 
yes. I cannot find it out that block is supported.
I guess it is only similar for attachments.
https://mattermost.atlassian.net/browse/MM-26729